src/test/java/org/benetech/servicenet/service/impl/ImportServiceImplIntTest.java

```diff
@@ -533,10 +1361,44 @@ private Organization generateNewOrganization(SystemAccount account) {
             .name(NEW_STRING).active(NEW_BOOLEAN);
     }
 
+    private Service generateExistingServiceDoNotPersist() {
+        return new Service().externalDbId(EXISTING_EXTERNAL_ID).providerName(PROVIDER)
+            .name(EXISTING_STRING);
+    }
+
     private Organization generateExistingOrganization(SystemAccount account) {
         Organization result = new Organization().externalDbId(EXISTING_EXTERNAL_ID).account(account)
             .name(EXISTING_STRING).active(EXISTING_BOOLEAN);
         em.persist(result);
         return result;
     }
+
+    private Organization generateExistingOrganizationDoNotPersist() {
+        return new Organization().externalDbId(EXISTING_EXTERNAL_ID).account(generateExistingAccount())
+            .name(EXISTING_STRING).active(EXISTING_BOOLEAN);
+    }
+
+    private Taxonomy generateNewTaxonomy() {
+        return new Taxonomy().externalDbId(NEW_EXTERNAL_ID).providerName(PROVIDER)
+            .name(NEW_STRING);
+    }
+
+    private Taxonomy generateExistingTaxonomy() {
+        Taxonomy result = new Taxonomy().externalDbId(EXISTING_EXTERNAL_ID).providerName(PROVIDER)
+            .name(EXISTING_STRING);
+        em.persist(result);
+        return result;
+    }
+
+    private Taxonomy generateOtherTaxonomy() {
+        Taxonomy result = new Taxonomy().externalDbId(EXISTING_EXTERNAL_ID).providerName(PROVIDER)
+            .name(OTHER_STRING);
+        em.persist(result);
+        return result;
+    }
+
+    @SafeVarargs
+    private <T> Set<T> mutableSet(T ... objects) {
+        return new HashSet<T>(Arrays.asList(objects));
+    }
```

Comment:
I'm afraid this class should be split into smaller ones, just like normal services :face_with_head_bandage:

Reply:
But we can definitely do this in a separate PR, but we have to remember it.